cpp/CMakeLists.txt

```diff
@@ -321,15 +321,26 @@ endif(PER_THREAD_DEFAULT_STREAM)
 
 if(BUILD_TESTS)
     include(CTest)
-    include(ConfigureGoogleTest)
+    find_package(GTest)
+    
+    if(NOT GTEST_ROOT)
+        include(ConfigureGoogleBenchmark)
+        set(GTEST_ROOT "${CMAKE_BINARY_DIR}/googletest")
+    endif(GTEST_ROOT)
+
+    add_subdirectory(${CMAKE_SOURCE_DIR}/tests)
+    
+    set(GTEST_INCLUDE_DIR "${GTEST_ROOT}/install/include")
+    set(GTEST_LIBRARY_DIR "${GTEST_ROOT}/install/lib")
+    link_directories(${GTEST_ROOT}/install/lib/)
+    include_directories(${GTEST_INCLUDE_DIR})
+
+    if(NOT GTEST_FOUND)
```

Comment:
Does `GTEST_FOUND` get set by the `ConfigureGoogleTest` script?

Reply:
Yes.